python/tests/kernel/test_kernel_features.py

```diff
@@ -2217,8 +2386,8 @@ def invalid_ctrl():
             rx.ctrl(np.pi, q)
 
         print(invalid_ctrl)
-    assert ('controlled operation requested without any control argument(s)'
-            in repr(error))
+    assert 'missing value' in repr(error)
```

Comment:
Isn't this diagnostic much less specific? Is that the intention?

Reply:
It is a bit less specific because all function arguments validations are now handled by the same logic. So while we can know the data types, we cannot know it's a control argument.